documentation/modules/auxiliary/admin/scada/advantech_webaccess_updatetemplate_sqli.md

```diff
@@ -0,0 +1,85 @@
+This module exploits a SQL injection vulnerability in Advantech WebAccess. The flaw can be found
+in the ```updateSubmit_Click``` function of the ```updateTemplate``` class, which is normally used for
+updating a *.xlsx template.
+
+The ```updateSubmit_Click``` function calls ```getTemplateIdByName``` from the ```DAO``` class
+(Data Access Object), which is actually where the SQL injection bug is:
+
+```
+// This line is from Page_Load
+// As you can see, template is user-controlled
+this.templateName = this.Request.QueryString["template"];
+
+... skipped ...
+
+protected void updateSubmit_Click(object sender, EventArgs e)
+{
+  DAO dao = new DAO(this.mdbPath, this.logPath);
+  int templateIdByName = dao.getTemplateIdByName(this.templateName);
+
+... skipped ...
+
+```
+
+The ```getTemplateIdByName``` function will execute a SQL statement with a user-supplied ```template```
+name, like so:
+
+```
+OleDbDataReader oleDbDataReader = new OleDbCommand("SELECT TemplateId FROM rRptTemplate WHERE TemplateName = '" + templateName + "'", this.conn).ExecuteReader();
+```
+
+Since there is no input validation for the ```template``` parameter, this can be exploited to
+inject additional SQL statements, and have data stolen. This module specifically steals
+credentials from existing project users stored in config/bwCfg.mdb.
+
+Although authentication is required, by default Advantech WebAccess uses a blank password for the
+built-in ```admin``` account, which allows the SQL injection vulnerability easier to exploit.
+
+## Vulnerable Application
+
+AdvanTech WebAccess 8.1 was specifically tested during development.
+
+To download this software for testing:
+
+http://advcloudfiles.advantech.com/web/Download/webaccess/8.1/AdvantechWebAccessUSANode8.1_20151230.exe
+
```

Comment:
Install on 2012 it didn't prompt me.  I think a note saying the software doesn't work on XP sp3 is good, and we'll leave out the need for the WWW server.